### PR TITLE
Adds project to wrap OTel SpanProcessor/SpanExporter as Money SpanHandler

### PR DESCRIPTION
**Pull request by HaloFour**
Enables wrapping an OTel SpanProcessor/SpanExporter as a Money SpanHandler so that Money can report spans directly to OpenTelemetry-compatible exporters, like ZipKin and Jaeger.

### CODE REVIEW

**Comment by HaloFour** [comment]
Working out the general shape of the API and then I plan on adding modules that wrap the exporters for Zipkin and Jaeger that can pull their configuration from TypeSafe.

**Comment by HaloFour** [comment]
woot, codecov report!

I'm going to take the WIP off of this PR.  I still have docs to add but I think functionally it's finished.

**Comment by HaloFour** [comment]
Added Scaladocs with sample config.  Also added an integration test for Zipkin.  Writing one for Jaeger would require using test containers which would be possible but a bit more involved.

File: money-otel-handler/src/test/java/com/comcast/money/otel/handlers/OtelSpanHandlerSpec.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        spanExporter = PowerMockito.mock(SpanExporter.class);
+        batchSpanProcessor = PowerMockito.mock(BatchSpanProcessor.class);
+        batchSpanProcessorBuilder = PowerMockito.mock(BatchSpanProcessor.Builder.class);
+        simpleSpanProcessor = PowerMockito.mock(SimpleSpanProcessor.class);
+        simpleSpanProcessorBuilder = PowerMockito.mock(SimpleSpanProcessor.Builder.class);
+
+        PowerMockito.when(BatchSpanProcessor.newBuilder(spanExporter)).thenReturn(batchSpanProcessorBuilder);
+        PowerMockito.when(SimpleSpanProcessor.newBuilder(spanExporter)).thenReturn(simpleSpanProcessorBuilder);
+        PowerMockito.when(batchSpanProcessorBuilder.build()).thenReturn(batchSpanProcessor);
+        PowerMockito.when(simpleSpanProcessorBuilder.build()).thenReturn(simpleSpanProcessor);
```
Comment (ptravers) [inline money-otel-handler/src/test/java/com/comcast/money/otel/handlers/OtelSpanHandlerSpec.java:72]: can power mock give false positives in your experience?

**Review by ptravers — APPROVED** [review]

**Comment by ptravers** [comment]
youve got conflicts?

File: money-otel-handler/src/test/java/com/comcast/money/otel/handlers/OtelSpanHandlerSpec.java (hunk 1 above)
Comment (HaloFour) [reply money-otel-handler/src/test/java/com/comcast/money/otel/handlers/OtelSpanHandlerSpec.java:72]: I haven't had any serious problems with it.  I hate reaching for it when it comes to Unit testing but a lot of the OTel APIs aren't mock-friendly, these builders are all `final` classes.

**Pull request merged**